net: pkt: Account for IP header length for allowable payload data

vakulgarg opened the pull request:
For calculating amount of payload data that can be added in a packet,
we need to subtract IPv6 or IPv6 header lengths from MTU.

Signed-off-by: Vakul Garg <vakul.garg@nxp.com>

pfl reviewed (approved):
Yes, indeed. Seems that header length got lost in the battle.
If I'd nitpick, i'd write 'data_len -=' instead.

nxa22042 committed:
net: pkt: Account for IP header length for allowable payload data

For calculating amount of payload data that can be added in a packet,
we need to subtract IPv6 or IPv4 header lengths from MTU.

Signed-off-by: Vakul Garg <vakul.garg@nxp.com>

pfalcon commented:
> The maximum data length that can be appended using net_pkt_append() should be set to TCP send_mss only if it is smaller than allowed payload length in net_pkt.

Can you explain what problem this patch fixes?

vakulgarg commented:
During test of http server sample using 6loble, I found tcp ctx send_mss
was set to 1220 (i.e ipv6_mtu - size of(v6_hdr) - size of(tcpheader).

However net_pkt->data_len was set to 1212 (since tcp opt hdr of 8 bytes) is
also considered. So variable max_len wrongly got increased to 1220 due to
unconditional assignment.



On 28-Nov-2017 5:49 PM, "Paul Sokolovsky" <notifications@github.com> wrote:

> The maximum data length that can be appended using net_pkt_append() should
> be set to TCP send_mss only if it is smaller than allowed payload length in
> net_pkt.
>
> Can you explain what problem this patch fixes?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/5179#issuecomment-347506115>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AM4O3GXKztp_FzTRh3o4PhUUlAOwSYZgks5s6_pPgaJpZM4Qstvp>
> .
>

nxa22042 committed:
net: pkt: Corrected data length appended in net_pkt_append()

The maximum data length that can be appended using net_pkt_append()
should be set to TCP send_mss only if it is smaller than allowed
payload length in net_pkt.

Signed-off-by: Vakul Garg <vakul.garg@nxp.com>
